python/ray/tests/test_runtime_env_complicated.py

```diff
@@ -648,6 +649,49 @@ def get(self):
     assert ray.get(worker_2.get.remote()) == (2, "2.3.0")
 
 
+@contextmanager
+def chdir(dir):
+    old_dir = os.getcwd()
+    os.chdir(dir)
+    yield
+    os.chdir(old_dir)
+
+
+@pytest.mark.skipif(
+    os.environ.get("CI") and sys.platform != "linux",
+    reason="This test is only run on linux CI machines.")
+def test_runtime_env_inheritance_regression(shutdown_only):
+    # https://github.com/ray-project/ray/issues/16479
+    with tempfile.TemporaryDirectory() as tmpdir, chdir(tmpdir):
+        with open("hello", "w") as f:
+            f.write("world")
+
+        job_config = ray.job_config.JobConfig(runtime_env={"working_dir": "."})
+        ray.init(job_config=job_config)
+
+        with open("hello", "w") as f:
+            f.write("file should already been cached")
+
+        @ray.remote
+        class Test:
+            def f(self):
+                return open("hello").read()
+
+        env1 = ray.get_runtime_context().runtime_env
+        del env1["working_dir"]
+        print("Using env:", env1)
+        t = Test.options(runtime_env=env1).remote()
+        assert ray.get(t.f.remote()) == "world"
+
+        # TODO(simon): This shows overriding working_dir has no effect.
+        # This tests the current behavior and the tests should change when
+        # we support per-task/actor runtime env.
+        env2 = ray.get_runtime_context().runtime_env
```

Comment:
Is there some consideration here that env2 is the same as env1 because we didn't deep-copy the dicts?  In other words does the `del` actually remove `"working_dir"` from the runtime_env in the job config (`ray.get_runtime_context().runtime_env`)?

Reply:
nope it seems like reconstructed each time https://github.com/ray-project/ray/blob/c691f73d8762a1e9038bd7de291b573ca8dc45a4/python/ray/worker.py#L188-L192